lib/std/tz.zig

```diff
@@ -0,0 +1,233 @@
+const std = @import("std.zig");
+const builtin = @import("builtin");
+
+pub const Transition = struct {
+    ts: i64,
+    timetype: *Timetype,
+};
+
+pub const Timetype = struct {
+    offset: i32,
+    flags: u8,
+    name_data: [6:0]u8,
+
+    pub fn name(self: Timetype) [:0]const u8 {
+        return std.mem.sliceTo(self.name_data[0..], 0);
+    }
+
+    pub fn isDst(self: Timetype) bool {
+        return (self.flags & 0x01) > 0;
+    }
+
+    pub fn standardTimeIndicator(self: Timetype) bool {
+        return (self.flags & 0x02) > 0;
+    }
+
+    pub fn utIndicator(self: Timetype) bool {
+        return (self.flags & 0x04) > 0;
+    }
+};
+
+pub const Leapsecond = struct {
+    occurrence: i48,
+    correction: i16,
+};
+
+pub const Tz = struct {
+    allocator: std.mem.Allocator,
+    transitions: []const Transition,
+    timetypes: []const Timetype,
+    leapseconds: []const Leapsecond,
+    footer: []const u8,
+
+    pub fn parse(allocator: std.mem.Allocator, reader: anytype) !Tz {
+        const Header = extern struct {
+            magic: [4]u8,
+            version: u8,
+            reserved: [15]u8,
+        };
+
+        const Counts = extern struct {
+            isutcnt: u32,
+            isstdcnt: u32,
+            leapcnt: u32,
+            timecnt: u32,
+            typecnt: u32,
+            charcnt: u32,
+        };
```

Comment:
These might be useful to hoist out of the function and have in the `struct`.

Reply:
Not sure they would be very useful for anything other than the function itself. I mostly added them as an optimization for the `reader`, so that it could fetch multiple values at the same time to reduce all the error branching.

Reply:
Yeah the only use-case I could think of was a tzfile writer.

Reply:
Done.